plugins/modules/zos_archive.py

```diff
@@ -0,0 +1,879 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) IBM Corporation 2020, 2022
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: zos_archive
+version_added: "1.7.0"
+author:
+  - Oscar Fernando Flores Garcia (@fernandofloresg)
+short_description: Archive a dataset on z/OS.
+
+description:
+  - Creates or extends an archive.
+  - The source and archive are on the remote host,
+    and the archive is not copied to the local host.
+
+options:
+  path:
+    description:
+      - Remote absolute path, glob, or list of paths or globs for the file or files to compress or archive.
+    type: list
+    required: true
+    elements: str
+  format:
+    description:
+      - The type of compression to use.
+    type: dict
+    required: false
+    suboptions:
+      name:
+        description:
+          - The name of the format to use.
+        type: str
+        required: false
+        default: gz
+        choices:
+          - bz2
+          - gz
+          - tar
+          - zip
+          - terse
+          - xmit
+          - pax
+      format_options:
+        description:
+          - Options specific to each format.
+        type: dict
+        required: false
+        suboptions:
+          terse_pack:
+            description: Pack option to use for terse format.
+            type: str
+            choices:
+              - PACK
+              - SPACK
+          xmit_log_dataset:
+            description: Provide a name of data set to use for xmit log.
+            type: str
+          use_adrdssu:
+            description:
+              - If set to true, after unpacking a data set in C(xmit) or C(terse) format
+                it will perform a single DFSMSdss ADRDSSU DUMP step.
+            type: bool
+            default: False
+  dest:
+    description: The file name of the dest archive.
```

Comment:
done

Reply:
Second line is missing a `-` hyphen, here is a copy/paste:
```
  dest:
    description:
      - The remote absolute path or data set where the archive should be created.
       - I(dest) can be a USS file or MVS data set name.
      - If I(dest) has missing parent directories, they will be created.
      - If I(dest) is a nonexistent USS file, it will be created.
      - Destination data set attributes can be set using I(dest_data_set).
    type: str
    required: false
```

I took out `- The module will default the size of destination file and destination data set.`  not sure what you decided on this, i think you are planning to take user input for the size if its a data set.

Reply:
Created the issue #891 to address the temporary data set sizes in case space attributes are not provided by the user.

Reply:
Description was updated.